files/usr/share/cinnamon/applets/sound@cinnamon.org/applet.js

```diff
@@ -150,6 +150,9 @@ VolumeSlider.prototype = {
             muted = true;
         } else {
             muted = false;
+            //100% is magnetic:
+            if (volume != this.applet._volumeNorm && volume > this.applet._volumeNorm*(1-VOLUME_ADJUSTMENT_STEP/2) && volume < this.applet._volumeNorm*(1+VOLUME_ADJUSTMENT_STEP/2))
```

Comment:
Do we need the check ```volume != this.appet._volumeNorm``` here? It will not change the value in any way, and it adds complexity to the condition, making it harder to read. I suppose it might avoid some calculations if the value is 100% already, but the calculations are pretty minor, so I can't imagine it would have a noticeable impact on performance.

Reply:
Yes, we need it. Without this test, which is done quickly because it compares two integers, there are two comparisons with floats that are made at each loop. If you try removing this test, you will notice a loss of performance and your fan will run out.

Reply:
Thanks for your tests, @collinss !